sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs

```diff
@@ -116,7 +116,7 @@ private OperationInternal(OperationState<T> finalState)
             _stateLock = new AsyncLockWithValue<OperationState<T>>(finalState);
         }
 
-        public override Response RawResponse => _stateLock.TryGetValue(out var state) ? state.RawResponse : _rawResponse;
+        public override Response RawResponse => _stateLock.TryGetValue(out var state) ? state.RawResponse : _rawResponse ?? throw new InvalidOperationException("The operation has not completed yet.");
```

Comment:
Not sure this error message reflects the state.  If `null` is passed as the raw response, you'll always trigger the exception.   No amount of waiting "for the operation to complete" will ever change that.

Reply:
I think this is similar to [OperationInternalOfT.Value](https://github.com/Azure/azure-sdk-for-net/blob/d49223f469c664319994ef18354c6e36e4325aca/sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs#L157), the user needs to call `UpdateStatus` ot `WaitForCompletion` to get the value.

Reply:
I think we should consider calling `UpdateStatus` in the Rehydate static method that way this will never be null and we won't need this throw.

This means we will want a sync Rehydrate and an async Rehydrate so that we can call the correct update status under the covers.

Reply:
Sounds good to me for rehydration.
For another case, the user can input null response for HLC custom constructor, should it call `UpdateStaus` firstly there as well?

Reply:
> This means we will want a sync Rehydrate and an async Rehydrate so that we can call the correct update status under the covers.

We are calling it in the constructor of [`RehydrationOperation`](https://github.com/Azure/azure-sdk-for-net/pull/42686/files#diff-2d5750d5b85a8da9948f6703c4dc97e3c9fbc72180173da74bfb16120370a556R22), can't make it async.